### PR TITLE
ENH: updates GNSS tec documentation, methods testing

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Improves the GNSS methods testing, adds a TEC example, and expands general Madrigal metadata definitions.  Also fully defines the GNSS VTEC metadata.  Partially addresses #3 and #41.

# Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality or documentation)
- This change requires a documentation update

# How Has This Been Tested?

Created an example in the docs and added unit tests.  You should also now receive no warning when loading VTEC data.

## Test Configuration
* Operating system: OS X Mojave
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: docs/examples/ex_gnss_tec.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   import numpy as np
+   import pysat
+   import pysatMadrigal as pysat_mad
+
+   vtec = pysat.Instrument(inst_module=pysat_mad.instruments.gnss_tec,
+                           tag='vtec')
+   ftime = dt.datetime(2013, 1, 1)
+
+   if not ftime in vtec.files.files.index:
+       vtec.download(start=ftime)
```
Comment (rstoneback) [inline docs/examples/ex_gnss_tec.rst:31]:
```suggestion
       vtec.download(start=ftime, user='firstname+lastname', password='myname@email.address')
```

File: docs/examples/ex_gnss_tec.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   ax.set_xlabel('Geographic Longitude ($^\circ$)')
+   ax.set_ylabel('Geodedic Latitude ($^\circ$)')
+   plt.colorbar(mappable=con, ax=ax, location='right', fraction=.1, pad=.01,
+                label='VTEC ({:s})'.format(vtec.meta['tec',
+		                                     vtec.meta.labels.units]))
+   fig.suptitle('{:s} {:s} at {:s}'.format(
+       vtec.platform.upper(), vtec.tag.upper(),
+       vtec.index[itime].strftime('%d %b %Y %H:%M:%S UT'), fontsize='medium'))
+
+
```
Comment (rstoneback) [inline docs/examples/ex_gnss_tec.rst:79]:
```suggestion
   plt.show()
```
or a `plt.ion()` earlier in the code.

File: pysatMadrigal/instruments/gnss_tec.py [hunk 3]
```diff
@@ -214,7 +214,19 @@ def load(fnames, tag=None, inst_id=None):
...
+        meta['dtec'] = {meta.labels.units: 'TECU', meta.labels.min_val: 0.0,
+                        meta.labels.max_val: np.nan}
+
+    # Get the maximum and minimum values for time, latitude, longitude,
+    # and altitude
+    meta['time'] = {meta.labels.notes: data['time'].values.dtype.__doc__,
+                    meta.labels.min_val: np.nan, meta.labels.max_val: np.nan}
+    meta['gdalt'] = {meta.labels.min_val: 0.0, meta.labels.max_val: np.nan}
+    meta['gdlat'] = {meta.labels.min_val: -90.0, meta.labels.max_val: 90.0}
+    min_lon = 0.0 if data['glon'].values.min() >= 0.0 else -180.0
```
Comment (rstoneback) [inline pysatMadrigal/instruments/gnss_tec.py:228]: Do the longitudes differ across files?

**Review by rstoneback — COMMENTED** [review]
Thanks @aburrell!

**Comment by rstoneback** [comment]
Not required, and it probably makes the example too complicated for the docs, but `it would be nice` if we could get coastlines on there as well.

File: pysatMadrigal/instruments/gnss_tec.py (hunk 3 above)
Comment (aburrell) [reply pysatMadrigal/instruments/gnss_tec.py:228]: Not yet... but it seemed silly to not check for something that varies regularly from file iteration to file iteration and is very simple to verify.

**Review by rstoneback — APPROVED** [review]
Thanks @aburrell!

**Pull request merged**